File: test/aplusb.cpp
// mle_example.cpp
#include <iostream>
#include <vector>
#include <sys/resource.h>

int main() {
    // 申请大约 3MB 的 vector<int>
    size_t n = 3 * 1024 * 1024 / sizeof(int);
    std::vector<int> v(n, 0);

    std::cout << "Allocated vector of size " << n << std::endl;

    struct rusage usage;
    getrusage(RUSAGE_SELF, &usage);
    double mem_mb = usage.ru_maxrss / 1024.0; // ru_maxrss 单位 KB
    std::cerr << "Memory used: " << mem_mb << " MB" << std::endl;

    return 0;
}

File: test/aplusb.py
# mle_example.py
import sys
import resource

# 申请大约 3MB 的列表
n = 3 * 1024 * 1024 // 8  # 3MB / 8 字节 per int
arr = [0] * n

# 输出到 stdout 正常信息
print(f"Allocated array of size {n}")

# 输出实际占用内存到 stderr
usage = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss / 1024  # MB
print(f"Memory used: {usage:.2f} MB", file=sys.stderr)
